static/js/context/auth.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, FC, ReactNode, useContext } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+const checkAuth = () =>
+  fetch("/1.0/certificates").then((response) => response.status !== 403);
```

Comment:
I think this should be in a dedicated file in API folder. Also this check fails on the demo server. Maybe it can be extended to also work there?

Reply:
Well, I don't know because I cannot do the add/remove thing from the lxc trusted list on the demo server, so I don't really have any way to test it...

Reply:
Right, maybe the better approach is to make the demo proxy return 403 for any calls in https://github.com/canonical/lxd-ui/blob/main/haproxy-demo.cfg#L12
If is_lxd_core is true and is_backend_allowed is false. If you feel like writing some HAProxy config feel free the add it, or I can look into it later on.

Reply:
I moved the solitary function to its own file.

About the demo server, I found a way to return 403 for any call and already tested it locally (= on the haproxy-dev config first), before pushing anything to the remote. However, I don't see how this can be so important since the demo server is already configured correctly and hence working without the auth check, so we don't really need to test this functionality on it. I don't even understand how you realised that it was failing there without manipulating the trust list.

However, I'm happy to test the method I found on the haproxy-demo config anyway, but out of the scope of this PR. I would either open another one later myself, expressly for this purpose, or you can feel free to check this issue later in a follow-up.

If there are no further remarks, I'd be happy to proceed with the merge. I can keep the PR unmerged until next week anyway, no problems with that.